version/distro/distro.go

```diff
@@ -47,6 +48,14 @@ func Get() Distro {
 	})
 }
 
+func IsWSL() bool {
+	return isWSL.Get(func() bool {
+		// We could look for $WSL_INTEROP instead, however that may be missing if
+		// the user has started to use systemd in WSL2.
+		return runtime.GOOS == "linux" && have("/proc/sys/fs/binfmt_misc/WSLInterop") || have("/mnt/wsl")
```

Comment:
The runtime.GOOS == "linux" part can go before line 52 (before the Get call) to deadcode more even if the IsWSL symbol makes it into code.

Reply:
FWIW I still see `WSL_INTEROP` with a vanilla Ubuntu distro with `systemd` enabled. 

Also:
`uname -r` => `5.15.90.1-microsoft-standard-WSL2`

Reply:
Good call!